packages/dev/core/src/Lights/IES/iesLoader.ts

```diff
@@ -0,0 +1,202 @@
+    const v = Lerp(v1, v2, t2);
+
+    return v;
+}
+
+/**
+ * Generates an IES texture from a string representing the IES data.
+ * @param source defines a string representing the IES data
+ * @param engine defines the engine to use
+ * @returns the IES texture
+ * @see https://ieslibrary.com/browse
+ * #UQGPDT
+ */
+export function LoadIESData(source: string, engine: AbstractEngine): RawTexture {
+    // Read data
+    const dataPointer = {
+        lines: source.split("\n"),
+        index: 0,
+    };
+    const data: IIESData = { version: dataPointer.lines[0], candelaValues: [], horizontalAngles: [], verticalAngles: [], numberOfHorizontalAngles: 0, numberOfVerticalAngles: 0 };
+
+    // Skip metadata
+    dataPointer.index = 1;
+    while (dataPointer.lines.length > 0 && !dataPointer.lines[dataPointer.index].includes("TILT=")) {
+        dataPointer.index++;
+    }
+
+    // Process tilt data?
+    if (dataPointer.lines[dataPointer.index].includes("INCLUDE")) {
+        // Not supported yet as I did not manage to find an example :)
+    }
+    dataPointer.index++;
+
+    // Header
+    const header = lineToArray(dataPointer.lines[dataPointer.index++]);
+    data.numberOfLights = header[0];
+    data.lumensPerLamp = header[1];
+    data.candelaMultiplier = header[2];
+    data.numberOfVerticalAngles = header[3];
+    data.numberOfHorizontalAngles = header[4];
+    data.photometricType = header[5]; // We ignore cylindrical type for now. Will add support later if needed
+    data.unitsType = header[6];
+    data.width = header[7];
+    data.length = header[8];
+    data.height = header[9];
+
+    // Additional data
+    const additionalData = lineToArray(dataPointer.lines[dataPointer.index++]);
+    data.ballastFactor = additionalData[0];
+    data.fileGenerationType = additionalData[1];
+    data.inputWatts = additionalData[2];
+
+    // Prepare arrays
+    for (let index = 0; index < data.numberOfHorizontalAngles; index++) {
+        data.candelaValues[index] = [];
+    }
+
+    // Vertical angles
+    readArray(dataPointer, data.numberOfVerticalAngles, data.verticalAngles);
+
+    // Horizontal angles
+    readArray(dataPointer, data.numberOfHorizontalAngles, data.horizontalAngles);
+
+    // Candela values
+    for (let index = 0; index < data.numberOfHorizontalAngles; index++) {
+        readArray(dataPointer, data.numberOfVerticalAngles, data.candelaValues[index]);
+    }
+
+    // Evaluate candela values
+    let maxCandela = -1;
+    for (let index = 0; index < data.numberOfHorizontalAngles; index++) {
+        for (let subIndex = 0; subIndex < data.numberOfVerticalAngles; subIndex++) {
+            data.candelaValues[index][subIndex] *= data.candelaValues[index][subIndex] * data.candelaMultiplier * data.ballastFactor * data.fileGenerationType;
+            maxCandela = Math.max(maxCandela, data.candelaValues[index][subIndex]);
+        }
+    }
+
+    // Normalize candela values
+    if (maxCandela > 0) {

```

Comment:
I wonder why we need to normalize if we rely on a float texture ? and if normalized, should we not keep the maxCandela maybe in the level property of the texture ?

Reply:
it is mostly to avoid the normalization in the shader (the photometric data will be in 0...1 range

Reply:
ohhhh I see !!! :-)